syft/workers/base.py

```diff
@@ -85,6 +85,8 @@ def __init__(
         self.verbose = verbose
         self.auto_add = auto_add
         self.msg_history = list()
+        self.promises = {}
+        self.tensor2promise_id = {}
```

Comment:
What is this parameter for?

Reply:
oops - bit early for that - this was originally a bigger PR but i had to stop because it was already getting quite large (better to merge along the way)